test/NuGet.Core.Tests/NuGet.ProjectModel.Test/PackageSpecCloningTests.cs

```diff
@@ -766,5 +766,23 @@ public void CompatibilityProfileCloneTest()
             Assert.Equal(3, compat.RestoreContexts.Count);
             Assert.Equal(2, clone.RestoreContexts.Count);
         }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void Clone_WhenIsDefaultVersionVaries_ReturnsEqualClone(bool expectedResult)
+        {
+            var packageSpec = new PackageSpec();
```

Comment:
Does the following example should work ?

```csharp
var pkgSpec = new PackageSpec();
pkgSpec.IsDefaultVersion = true;
pkgSpec.Version = new Versioning.NuGetVersion("1.0.0");
var cln = pkgSpec.Clone();

Assert.True( cln.Version.Equals(obj.Version) );
Assert.True( cln.IsDefaultVersion == cln.IsDefaultVersion ); 
```

Reply:
That should have worked before and after this change because this statement:
```C#
pkgSpec.Version = new Versioning.NuGetVersion("1.0.0");
```
…[undoes](https://github.com/NuGet/NuGet.Client/blob/f67a6083f8a49502f52f0c50492db03452785359/src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs#L49) the previous statement:
```C#
pkgSpec.IsDefaultVersion = true;
```
...and both the original and the clone would have `IsDefaultVersion` set to `false`.